src/uu/numfmt/src/format.rs

```diff
@@ -85,18 +81,27 @@ fn parse_suffix(s: &str) -> Result<(f64, Option<Suffix>)> {
         Some('R') => Some((RawSuffix::R, with_i)),
         Some('Q') => Some((RawSuffix::Q, with_i)),
         Some('0'..='9') if !with_i => None,
-        _ => {
+        Some(invalid_suffix) => {
             // If with_i is true, the string ends with 'i' but there's no valid suffix letter
             // This is always an invalid suffix (e.g., "1i", "2Ai")
             if with_i {
                 return Err(translate!("numfmt-error-invalid-suffix", "input" => s.quote()));
             }
             // For other cases, check if the number part (without the last character) is valid
             let number_part = &s[..s.len() - 1];
-            if number_part.is_empty() || number_part.parse::<f64>().is_err() {
-                return Err(translate!("numfmt-error-invalid-number", "input" => s.quote()));
-            }
-            return Err(translate!("numfmt-error-invalid-suffix", "input" => s.quote()));
+
+            return Err(match parse_suffix(number_part) {
```

Comment:
Recursive call to parse_suffix() in error path - consider adding a comment explaining this won't cause stack overflow due to guaranteed termination